crates/red_knot_python_semantic/src/semantic_index/symbol.rs

```diff
@@ -322,6 +384,14 @@ impl SymbolTableBuilder {
     }
 }
 
+impl Deref for SymbolTableBuilder {
+    type Target = SymbolTable;
+
+    fn deref(&self) -> &Self::Target {
+        &self.table
+    }
+}
+
```

Comment:
Hmm, given that `self.table` on a `SymbolTableBuilder` is a potentially incomplete symbol table, it seems a little risky to me to implement this transparent `Deref`. Where in this diff do we rely on that?

Reply:
I think this is probably my fault 😶

IIRC, I added this as a "quick and easy solution" to get a scrappy version working, intending to remove it later. But I think I forgot to tell Doug that. Sorry @dcreager!

Reply:
(It looks like this patch _does_ rely on a couple of the methods from the `SymbolTable` in `builder.rs` currently, though)